### ISSUE TITLE
dump issue with one conformance file: file9.jp2 

### ISSUE THREAD

**Issue by gcode-importer** [uclouvain/openjpeg#96]
Originally reported on Google Code with ID 96
```
What steps will reproduce the problem?
1. j2k_dump -i file9.jp2 -o dump-file9.jp2-1.txt
2. j2k_dump -i file9.jp2 -o dump-file9.jp2-2.txt
3. compare the output files

Dump is correctly done but output is variable. More precisely stepsize values are different.




```

Reported by [savmickael](https://github.com/savmickael) on 2011-10-07 13:45:34

**Comment by detonin** [comment]
Bug still present
dump output is unstable.

````226/1207 Testing: NR-file9.jp2-compare_dump2base
226/1207 Test: NR-file9.jp2-compare_dump2base
Command: "/Users/antonin/dev/opj/github/openjpeg15-build/bin/compare_dump_files" "-b" "/Users/antonin/dev/opj/github/openjpeg-data/baseline/nonregression/opj_file9.txt" "-t" "/Users/antonin/dev/opj/github/openjpeg15-build/tests/conformance/Temporary/file9.jp2.txt"
Directory: /Users/antonin/dev/opj/github/openjpeg15-build/tests/conformance
"NR-file9.jp2-compare_dump2base" start time: Sep 17 10:01 CEST
Output:
----------------------------------------------------------
<      numgbits=0> vs. <      numgbits=147865536>
******Parameters********* 
 base_filename = /Users/antonin/dev/opj/github/openjpeg-data/baseline/nonregression/opj_file9.txt
 test_filename = /Users/antonin/dev/opj/github/openjpeg15-build/tests/conformance/Temporary/file9.jp2.txt
************************* 
Try to open: /Users/antonin/dev/opj/github/openjpeg-data/baseline/nonregression/opj_file9.txt for reading ... Ok.
Try to open: /Users/antonin/dev/opj/github/openjpeg15-build/tests/conformance/Temporary/file9.jp2.txt for reading ... Ok.
<end of output>
Test time =   0.00 sec
----------------------------------------------------------
Test Failed.
"NR-file9.jp2-compare_dump2base" end time: Sep 17 10:01 CEST
"NR-file9.jp2-compare_dump2base" time elapsed: 00:00:00
----------------------------------------------------------
```

**Comment by detonin** [comment]
same problem with issue104_jpxstream.jp2

**Comment by detonin** [comment]
j2k codestream in file9.jp2 has one component but decoder output is 3 components thanks to a component mapping box and a palette box in jp2 structure.
So decoded file shall be 3 components but dumped file shall refer to only one component (j2k_dump only analyzes the codestream, not the jp2 file format).
Same problem with issue104_jpxstream.jp2

**Comment by detonin** [comment]
fixed by 32b04ad49d1fc17bf8e9b6a1725a465f57b7adfb
